packages/accordion/src/useAccordion.ts

```diff
@@ -5,185 +5,144 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import { useState, HTMLProps, useMemo } from 'react';
-import { useUIDSeed } from 'react-uid';
+import { useState, useMemo, useCallback } from 'react';
 import {
-  composeEventHandlers,
   KEY_CODES,
-  getControlledValue
+  composeEventHandlers,
+  getControlledValue,
+  useId
 } from '@zendeskgarden/container-utilities';
 
-export interface IUseAccordionProps {
-  /** Prefixes IDs for the accordion trigger and panels  */
-  idPrefix?: string;
-  /** Sets the expanded sections in a controlled accordion */
-  expandedSections?: number[];
-  /** Sets the default expanded sections in a uncontrolled accordion */
-  defaultExpandedSections?: number[];
-  /** Handles accordion expansion changes */
-  onChange?: (expanded: number) => any;
-  /** Determines if multiple panels can be expanded at the same time in an uncontrolled accordion */
-  expandable?: boolean;
-  /** Determines if panels can be collapsed in an uncontrolled accordion */
-  collapsible?: boolean;
-}
-
-interface IHeaderProps extends HTMLProps<any> {
-  ariaLevel?: number | null;
-  role?: any;
-}
-
-interface ITriggerProps extends HTMLProps<any> {
-  index?: number;
-  role?: any;
-  tabIndex?: any;
-}
-
-interface IPanelProps extends HTMLProps<any> {
-  index?: number;
-  role?: any;
-}
-
-export interface IUseAccordionPropGetters {
-  getHeaderProps: <T>(options?: T & IHeaderProps) => any;
-  getTriggerProps: <T>(options?: T & ITriggerProps) => any;
-  getPanelProps: <T>(options?: T & IPanelProps) => any;
-}
-
-export interface IUseAccordionReturnValue extends IUseAccordionPropGetters {
-  expandedSections: number[];
-  disabledSections: number[];
-}
+import { IUseAccordionProps, IUseAccordionReturnValue } from './types';
 
-export function useAccordion({
+export function useAccordion<Value>({
   idPrefix,
+  sections,
```

Comment:
```suggestion
  sections = [],
```

Reply:
Sections will be required, what would be the reason behind providing a default value?

Reply:
As a rule, we should add sensible defaults for required hook parameters. The `PropTypes` help enforce for component JSX usage; but this mitigates foot-gun for hook usage. Reference https://github.com/zendeskgarden/react-containers/blob/8a30d4b802ae3042b5f1abfffac39683f2dedb35/packages/combobox/src/useCombobox.ts#L38

Reply:
Updated!